### PR TITLE
Fermi2D: limit blit area to only available area

### PR DESCRIPTION
**Pull request by FernandoS27**
Normaly OpenGL does not care if the areas exceed the texture regions but other backends such as Vulkan do care about the limits of this areas. This PR crops the areas of the blit in order that they don't surpass the limits of the textures. This should help Vulkan and faulty OpenGL drivers.

### CODE REVIEW

**Comment by ReinUsesLisp** [comment]
I'm not sure this is the right approach. `src` and `dest` are off by one pixel, e.g.: `[-1,1279]`. It looks more like an issue in the code above these changes.

**Comment by FernandoS27** [comment]
> I'm not sure this is the right approach. `src` and `dest` are off by one pixel, e.g.: `[-1,1279]`. It looks more like an issue in the code above these changes.

I could look at my logical, it could also mean our base equations may be wrong. Some hardware tests could do.

File: src/video_core/engines/fermi_2d.cpp [hunk 1]
```diff
@@ -28,6 +28,13 @@ void Fermi2D::CallMethod(const GPU::MethodCall& method_call) {
     }
 }
 
+std::pair<u32, u32> DelimitLine(u32 src_1, u32 src_2, u32 dst_1, u32 dst_2, u32 src_line) {
```
Comment (ReinUsesLisp) [inline src/video_core/engines/fermi_2d.cpp]: This should go in an anonymous namespace, at the top of the source file.

**Comment by bunnei** [comment]
@FernandoS27 is this OK to be merged?

**Comment by bunnei** [comment]
@FernandoS27 ping again

**Comment by FernandoS27** [comment]
@bunnei yeah

**Pull request merged**